### PR TITLE
[backend] feat: return not_found in refresh when platform is not found

### PR DESCRIPTION
**Pull request by Kroustille**
# Context
> Briefly describe the purpose and context of this PR. Include relevant screenshots or screen recordings for the product team.

We still have an issue where OpenCTI sends connectivity lost emails for training platforms. There are created from a template but with a different platform id. This id is unknown from the hub, OpenCTI gets an inactive status, and we send the connectivity lost email.

This PR allows OpenCTI to know when the platform id is not registered in the hub to prevent this email.

- call the mutation with a platform id which does not corresponds to an existing platform, and with a `platform_version` greater than required version
- check it returns `not_found`
- call the mutation with a platform id which does not corresponds to an existing platform, and with a `platform_version` lower than required version
- check it returns `inactive`

### OpenCTI

https://github.com/user-attachments/assets/c1d56236-e544-429e-aa15-e168d700daf1

### OpenAEV

https://github.com/user-attachments/assets/1028536e-4625-4195-93ee-2fb1abb6929c

## Related issues

- Related to #1286

# Checklist

## Quality

- [x] I consider this work complete and ready for review
- [x] I tested all features and edge cases
- [x] I refactored code where necessary to improve quality
- [x] I made sure my code meets development guidelines
- [x] I performed a self-review of my code

## Testing

- [x] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [ ] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [ ] Unit tests
- [x] Integration tests
- [ ] E2E tests
- [x] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

File: apps/portal-api/src/modules/services/registration/registration.app.test.ts [hunk 1]
```diff
@@ -636,12 +636,50 @@ describe('Registration app', () => {
...
+          registrationApp.refreshPlatformRegistrationConnectivityStatus({
+            platformId: uuidv4(),
+            token: uuidv4(),
+            platformVersion: '9.Y.Z',
+          });
+
+        await expect(call).rejects.toThrow(ErrorCode.InvalidPlatformVersion);
+      });
+
+      it('should return inactive when platform is not registered but identifier is not provided', async () => {
```
[the remark this answers is not in the document]
Comment (Kroustille) [reply apps/portal-api/src/modules/services/registration/registration.app.test.ts]: It's to keep compatibility with older OpenCTI or OpenAEV versions which don't send platform identifier.

**Review by jpkha — APPROVED** [review]
